Utilities/Loader.cpp

```diff
@@ -15,7 +36,144 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
-//#include <experimental/filesystem>
+#include <string.h> // strcmp
+#include <algorithm> // for split
+#include <iterator> // for split
+
+std::string dataVersionOfFilename(const std::string& filename) {
+    std::string ext(filename.substr(filename.rfind(".")));
+    int orgposIfExist = filename.rfind("organisms");
+    if (orgposIfExist != -1) return filename.substr(0,orgposIfExist)+"data"+filename.substr(orgposIfExist+9);
+    else return filename.substr(0,filename.rfind("."))+"_data"+ext;
+}
+
+bool fileExists(const std::string& filename) {
+#if defined(OS_UNIX)
+    struct stat statbuf; // linux only
+    return (stat(filename.c_str(), &statbuf) == 0);
+#elif defined(OS_WINDOWS)
+    DWORD fileAttr;
+    fileAttr = GetFileAttributesA(filename.c_str());
+    return (0xFFFFFFFF != fileAttr);
+#endif
+}
+
+bool isDirectory(const std::string& dirname) {
+#if defined(OS_UNIX)
+    struct stat statbuf; // linux only
+    stat(dirname.c_str(), &statbuf);
+	 return (S_ISDIR(statbuf.st_mode));
+#elif defined(OS_WINDOWS)
+    DWORD fileAttr;
+    fileAttr = GetFileAttributesA(dirname.c_str());
+	 return (fileAttr & FILE_ATTRIBUTE_DIRECTORY);
+#endif
+}
+
+template <class Container>
+void split(const std::string& str, Container& cont, char delim = ' ') {
+    std::stringstream ss(str);
+    std::string token;
+    while (std::getline(ss, token, delim)) {
+        cont.push_back(token);
+    }
+}
+
+bool wildstrcmp(const char *first, const char *second) {
```

Comment:
This function should simply be a regex.

Reply:
At first glance, this seems like a good idea. However, the user string (the `first` parameter in the fn) is not the same as a regex string and would need to be appropriately transformed. We can do it, but need to make sure to catch everything, which is kind of messy.
* `?` needs to become `.?`
* `*` needs to become `.*`
* `\` needs to become `\\`
* then...
* `.` needs to become `\.` (maybe `\\.` in c++)
What else is missing? Either way, there is some dirty stuff that needs to happen, so I propose what's there is fine because it works.

Reply:
We can decide what kind of wildcards the user is allowed, and only account for that. The code for that is essentially the commented lines 684-686 in this PR. 

While the current implementation works, it's fairly unreadable. Any future modifications will be a pain to do correctly.